domain/src/coaching_session.rs

```diff
@@ -70,6 +71,29 @@ pub async fn find_by(
     Ok(coaching_sessions)
 }
 
+pub async fn find_by_with_sort<P>(db: &DatabaseConnection, params: P) -> Result<Vec<Model>, Error>
+where
+    P: IntoQueryFilterMap + CoachingSessionSortParams,
+{
+    // Extract sort parameters before consuming params
+    let sort_column = params.get_sort_column();
+    let sort_order = params.get_sort_order();
+    let query_filter_map = params.into_query_filter_map();
+
+    let coaching_sessions = query::find_by_with_sort::<
+        coaching_sessions::Entity,
+        coaching_sessions::Column,
+    >(db, query_filter_map, sort_column, sort_order)
+    .await?;
+
+    Ok(coaching_sessions)
+}
+
+pub trait CoachingSessionSortParams {
```

Comment:
This trait (or at least the concept and functionality), I propose, belongs in `entity_api/src/query.rs` and should be made to be generic in the same way that `IntoQueryFilterMap` is used. That way other entities can be made sortable.